Выполнила ФЗ 9

lapochka2019 opened the pull request.

lapochka2019 committed:
Выполнила ФЗ 9

lapochka2019 committed:
Исправила ошибки стиля (1)

lapochka2019 committed:
Исправила ошибки стиля (2)

vlolad commented on src/Main.java:
```diff
@@ -1,10 +0,0 @@
-import model.*;
-import service.*;
-
-//Нужен ли теперь это класс?
```
Ну, главное что без метода main приложения не получится, а он уже есть в другом классе)
Можно удалить, можно оставить, up to you

vlolad commented on src/server/HttpTaskServer.java:
```diff
@@ -0,0 +1,59 @@
+        HttpTaskServer.manager = manager;
+        server = HttpServer.create(new InetSocketAddress(PORT), 5);
+
+        server.createContext("/tasks", new TaskHandler(gson, manager));
```
Отлично, что разделены хендлеры по разным классам

vlolad commented on src/server/hadler/BaseHttpHandler.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    public boolean isGet(String methodType) {
+        return "GET".equals(methodType);
```
Хорошая идея с отдельными методами для проверки. Можно ещё строки запросов в Enum упаковать, но не обязательно

vlolad commented on src/server/hadler/model/EpicHandler.java:
```diff
@@ -0,0 +1,102 @@
+        String methodType = exchange.getRequestMethod();
+        Optional<Integer> epicId = getId(exchange);
+        //"/epics"
+        if (epicId.isEmpty()) {
```
Очень разбух код хендлера. Логику анализа звапроса надо оставить тут, а саму обработку (которая занимает больше одной строки) - вынести в отдельные методы. 

Например, то что описано после условия `isPost(methodType)`

vlolad commented on src/server/hadler/model/SubtaskHandler.java:
```diff
@@ -0,0 +1,87 @@
+        String methodType = exchange.getRequestMethod();
+        Optional<Integer> subtaskId = getId(exchange);
+        //"/subtasks"
+        if (subtaskId.isEmpty()) {
```
Аналогично комментарию в EpicHandler. Код разбух, надо разбить по отдельным методам

vlolad commented on src/server/hadler/model/TaskHandler.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
```
Надо разбить по отдельным методам

vlolad commented on src/service/InMemoryHistoryManager.java:
```diff
@@ -17,11 +17,11 @@ public ArrayList<Task> getHistory() {
+        if (task == null) {
+            return;
+        }
+        //Получается я тут в любом случае вызываю метод удаления?
```
Не понял вопроса)
Если task == null, то метод удаления не вызвать, потому что `task.getId()` вызовет NPE

vlolad commented on src/service/InMemoryTaskManager.java:
```diff
@@ -17,27 +18,32 @@ public class InMemoryTaskManager implements TaskManager {
     protected int idCount = 0;
-    //пришлось убрать приватность для теста
-    InMemoryHistoryManager history;
+    protected InMemoryHistoryManager history;
```
protected подходит

vlolad reviewed (changes requested):
Привет!

По функционалу всё хорошо, но хендлеры для типов задач надо рефакторить. Один огромный метод handle во всех. Надо разбить на отдельные методы, куда вынести логику обработки тех или иных запросов

В остальном только пожелания по улучшению)

Хорошего дня!

lapochka2019 committed:
Проведен рефакторинг хендлеров
Создан класс Main для реализации метода main (чтобы не пихать этот метод куда попало)

lapochka2019 committed:
Исправила ошибки стиля (1)

lapochka2019 committed:
Исправила ошибки стиля (2)

lapochka2019 committed:
Исправила ошибки стиля (3)

vlolad reviewed (approved):
Привет!

Всё принято, поздравляю с завершением первого большого проекта!
В дальнейшем настраивать HTTP-сервер вручную больше  не придется, но скучно точно не будет с:

Хорошей недели!
